flink-runtime/src/test/java/org/apache/flink/runtime/util/config/memory/ManagedMemoryUtilsTest.java

```diff
@@ -132,27 +131,33 @@ public void testGetWeightsFromConfigLegacy() {
                 ManagedMemoryUtils.getManagedMemoryUseCaseWeightsFromConfig(
                         CONFIG_WITH_LEGACY_USE_CASES);
 
-        assertThat(configuredWeights, is(expectedWeights));
+        assertThat(configuredWeights).isEqualTo(expectedWeights);
     }
 
-    @Test(expected = IllegalConfigurationException.class)
-    public void testGetWeightsFromConfigFailNegativeWeight() {
-        final Configuration config =
-                new Configuration() {
-                    {
-                        set(
-                                TaskManagerOptions.MANAGED_MEMORY_CONSUMER_WEIGHTS,
-                                Collections.singletonMap(
-                                        TaskManagerOptions.MANAGED_MEMORY_CONSUMER_NAME_OPERATOR,
-                                        "-123"));
-                    }
-                };
+    @Test
+    void testGetWeightsFromConfigFailNegativeWeight() {
+        assertThatThrownBy(
+                        () -> {
+                            final Configuration config =
+                                    new Configuration() {
+                                        {
+                                            set(
+                                                    TaskManagerOptions
+                                                            .MANAGED_MEMORY_CONSUMER_WEIGHTS,
+                                                    Collections.singletonMap(
+                                                            TaskManagerOptions
+                                                                    .MANAGED_MEMORY_CONSUMER_NAME_OPERATOR,
+                                                            "-123"));
+                                        }
+                                    };
```

Comment:
The config can be removed from this block, right?

The expected `IllegalConfigurationException` should be thrown from `ManagedMemoryUtils.getManagedMemoryUseCaseWeightsFromConfig(config);`.